test/unit/teacher_set_test.rb

```diff
@@ -182,12 +186,218 @@ class TeacherSetTest < ActiveSupport::TestCase
+      bibd_id = SIERRA_USER["data"][0]['id']
+
+      ts_items_info = {bibs_resp: SIERRA_USER, total_count: 1, available_count: 1, availability_string: 'available'}
+      
+      TeacherSet.stub_any_instance :get_items_info_from_bibs_service, ts_items_info do
+        TeacherSet.stub_any_instance :create_or_update_teacherset_document_in_es, es_resp do
+          resp = TeacherSet.create_or_update_teacher_set(SIERRA_USER["data"][0])
+        end
+      end
+      assert_equal(@teacher_set6.bnumber, resp.bnumber)
+    end
+  end
+
+  # Update teacher-set document in ES
+  describe '#test create or update teacherset' do
+    before do
+      @es_doc = {"_index" => "teacherset", "_type" => "teacherset", 
+                "_id" => 8888, "_version" => 11, "result" => "updated", 
+                "_shards" => {"total" => 0, "successful" => 1, "failed" => 0}}
+      @expected_resp = {:title=>"title",:description=>"desc",:contents=>nil,:id=>8888,:details_url=>nil,
+                        :grade_end=>nil,:grade_begin=>nil,:availability=>nil,:total_copies=>nil,
+                        :call_number=>nil,:language=>nil,:physical_description=>nil,:primary_language=>nil,
+                        :created_at=>nil,:updated_at=>nil,:available_copies=>nil,
+                        :bnumber=>"bnumber",:set_type=>nil,:area_of_study=>nil,:subjects=>[]}
+    end
+
+    it 'update teacher-set document in ElasticSearch' do
+      resp = nil
+      teacher_set = TeacherSet.new(title: 'title', description: 'desc', bnumber: "bnumber", id: "8888")
+      elasticsearch_adapter_mock = Minitest::Mock.new
+      elasticsearch_adapter_mock.expect(:update_document_by_id, @es_doc, [teacher_set.id, @expected_resp])
+      ElasticSearch.stub :new, elasticsearch_adapter_mock do
+        resp = teacher_set.create_or_update_teacherset_document_in_es
+      end
+      assert_equal(teacher_set.id.to_i, resp["_id"])
+      elasticsearch_adapter_mock.verify
+    end
+
+
+    it 'raises ElasticsearchException when ElasticSearch raises StandardError' do
+      detail_msg = 'exception occured'
+      teacher_set = TeacherSet.new(title: 'title', description: 'desc', bnumber: "bnumber", id: "8888")
+      elasticsearch_adapter_mock = Minitest::Mock.new
+      elasticsearch_adapter_mock.expect(:update_document_by_id, nil) do
+        raise StandardError, ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg]
+      end
+    
+      resp = assert_raises(ElasticsearchException) do
+        ElasticSearch.stub :new, elasticsearch_adapter_mock do
+          teacher_set.create_or_update_teacherset_document_in_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+      elasticsearch_adapter_mock.verify
+    end
+  end
+
+
+  # Delete teacher-set document in ES
+  describe "Delete teacherset record from elastic search" do
```

Comment:
So a good practice for these descriptions is to say what method is being tested in the `describe`, and then say what you are specifically testing in the `it` line. 

For example:

```
  describe "TeacherSet#delete_teacherset_record_from_es" do
      it "should call ElasticSearch#delete_document_by_id"
          # test here
```